### ISSUE TITLE
Implement generate_merfish_comparisons.py

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#191]
## Summary

Create `tests/visual_eval/generate_merfish_comparisons.py` — the MERFISH-specific
UMAP comparison pipeline. This is the script that `run_merfish_eval.sh` already calls
(`--phase baseline` and `--phase compare`) but which does not yet exist. Once
implemented, the full three-phase evaluation pipeline (Python baseline → Rust export →
Python comparison) will be runnable end-to-end on the 10K subset.

## Context

- Plan reference: `docs/merfish-umap-benchmark-plan.md` Batch 3, Issue 3.1
- The shell orchestrator `run_merfish_eval.sh` calls this script in Phases 1 and 3
- The Rust test `export_merfish_init.rs` reads `merfish_10k_graph.npz` and writes
  `merfish_10k_rust_init.npy` — this script must produce the former and consume the latter
- The 10K subset is committed at `tests/visual_eval/merfish_data/` (compressed `.npz`)
- Must follow the exact patterns from `generate_umap_comparisons.py` for graph export
  format, metric computation, and plot generation

### Phase: baseline (`--phase baseline`)

1. **Load 10K subset** from `tests/visual_eval/merfish_data/`:
   - `merfish_10k_expression.npz` → float32 (10000, 1122) expression matrix
   - `merfish_10k_spatial.npz` → float32 (10000, 2) section (x, y) coordinates
   - `merfish_10k_labels.npz` → int32 (10000,) cluster_alias cell-type IDs
   - `merfish_10k_section_ids.npz` → int32 (10000,) brain section indices

2. **Preprocess** (following Zhuang lab conventions, plan Section 6.1):
   - `sc.pp.normalize_total(adata, target_sum=1000)`
   - `sc.pp.log1p(adata)`
   - `sc.pp.scale(adata, max_value=10)`
   - `sc.tl.pca(adata, n_comps=30)` (start with 30; RQ-1.2 may adjust later)

3. **Build kNN graph and fit UMAP:**
   - `sc.pp.neighbors(adata, n_neighbors=15, n_pcs=30, metric='euclidean')`
   - `umap.UMAP(n_neighbors=15, min_dist=0.1, n_components=2, metric='euclidean',
     random_state=42, n_jobs=1)`
   - Run `umap.spectral.spectral_layout()` for Python spectral init coordinates

4. **Export graph** via `export_graph()` — exact same implementation as
   `generate_umap_comparisons.py` (f32 data, i32 indices, i32 indptr, i32 shape).
   This is what `export_merfish_init.rs` loads.

5. **Save baseline artifacts** to `tests/visual_eval/output/`:
   - `merfish_10k_graph.npz` — kNN graph (CSR, for Rust)
   - `merfish_10k_py_spectral.npy` — Python spectral init coords
   - `merfish_10k_py_final.npy` — Python UMAP final embedding
   - `merfish_10k_labels.npy` — labels (copy for compare phase)
   - `merfish_10k_spatial.npz` — spatial coordinates (for future Category B metrics)
   - `merfish_10k_baseline.png` — 2×2 plot (spectral init, final embed, eigenvalues, metrics)

### Phase: compare (`--phase compare`)

1. **Load artifacts** from `output/`:
   - `merfish_10k_rust_init.npy` (written by Rust export)
   - `merfish_10k_py_spectral.npy`, `merfish_10k_py_final.npy`, `merfish_10k_labels.npy`

2. **Run three-way UMAP SGD** (identical UMAP params):
   - Python spectral → SGD (already computed, reuse `py_final`)
   - Rust spectral → SGD: `UMAP(init=rust_init, ...).fit_transform(X)`
   - Random → SGD: `UMAP(init='random', ...).fit_transform(X)`

3. **Compute Category A metrics** via `_compute_metrics()`:
   - Trustworthiness (delta < 0.01)
   - Silhouette (delta < 0.05)
   - Procrustes disparity (< 0.05)
   - Pairwise distance correlation (> 0.99)
   - Overall PASS/FAIL

4. **Generate plots:**
   - `merfish_10k_comparison.png` — 2×3 grid (pre-SGD top, post-SGD bottom)
   - `merfish_10k_overlay.png` — Python vs Rust overlay
   - `merfish_10k_three_way_overlay.png` — Python vs Rust vs Random

5. **Write** `merfish_10k_metrics.json` — same schema as existing metrics files

### CLI

```bash
python tests/visual_eval/generate_merfish_comparisons.py --phase baseline
python tests/visual_eval/generate_merfish_comparisons.py --phase compare
```

No `--dataset` or `--tier` arguments needed — this script handles only `merfish_10k`.

## Critical implementation details

- **Graph format must exactly match** what `export_merfish_init.rs` expects:
  `data=f32, indices=i32, indptr=i32, shape=i32` via `np.savez()`
- **UMAP params must match** `generate_umap_comparisons.py`: `n_neighbors=15,
  min_dist=0.1, n_components=2, metric='euclidean', random_state=42, n_jobs=1`
- **Metric thresholds must match**: Procrustes < 0.05, pairwise_corr > 0.99,
  trustworthiness delta < 0.01, silhouette delta < 0.05
- **Plot style must match**: `matplotlib.use('Agg')`, dpi=150, seaborn-v0_8-whitegrid
- Preprocessing uses `scanpy` (normalize_total, log1p, scale, pca, neighbors)
- The expression data needs to be loaded into an AnnData object for scanpy preprocessing
### LOAD — Data Loading

- **REQ-LOAD-001:** The script must load the 10K subset from `tests/visual_eval/merfish_data/` using compressed `.npz` files via `np.load()`.
- **REQ-LOAD-002:** The expression matrix must be loaded as float32 with shape (10000, 1122).
- **REQ-LOAD-003:** Spatial coordinates must be loaded and carried through both phases, saved as `merfish_10k_spatial.npz` in the output directory.

### PREP — Preprocessing

- **REQ-PREP-001:** Preprocessing must use scanpy: `normalize_total(target_sum=1000)` → `log1p()` → `scale(max_value=10)` → `pca(n_comps=30)`.
- **REQ-PREP-002:** The expression data must be wrapped in an AnnData object for scanpy compatibility.
- **REQ-PREP-003:** The kNN graph must be built with `n_neighbors=15`, `n_pcs=30`, `metric='euclidean'`.

### GRAPH — Graph Export

- **REQ-GRAPH-001:** The kNN graph must be exported via `export_graph()` with dtypes: data=float32, indices=int32, indptr=int32, shape=int32.
- **REQ-GRAPH-002:** The exported graph must be loadable by `export_merfish_init.rs` via the `load_sparse_csr_f32_u32` helper without modification.
- **REQ-GRAPH-003:** The graph must be saved as `merfish_10k_graph.npz` in `tests/visual_eval/output/`.

### UMAP — UMAP Parameters

- **REQ-UMAP-001:** All UMAP runs must use `n_neighbors=15, min_dist=0.1, n_components=2, metric='euclidean', random_state=42, n_jobs=1`.
- **REQ-UMAP-002:** The baseline phase must compute both Python spectral init coordinates (via `spectral_layout`) and the final SGD embedding.
- **REQ-UMAP-003:** The compare phase must run three SGD embeddings: from Python spectral init, from Rust spectral init, and from random init.

### METRIC — Metrics and Output

- **REQ-METRIC-001:** The `_compute_metrics()` function must use identical thresholds to `generate_umap_comparisons.py`: Procrustes < 0.05, pairwise_corr > 0.99, trustworthiness delta < 0.01, silhouette delta < 0.05.
- **REQ-METRIC-002:** The output `merfish_10k_metrics.json` must follow the same schema as existing `*_metrics.json` files (keys: `python_spectral`, `rust_spectral`, `random`, `pass_fail`, `dataset`, `n_samples`, `n_features`).

### PLOT — Visualization

- **REQ-PLOT-001:** The baseline phase must produce `merfish_10k_baseline.png` (2×2 grid: spectral init, final embed, eigenvalue spectrum, metrics text).
- **REQ-PLOT-002:** The compare phase must produce `merfish_10k_comparison.png`, `merfish_10k_overlay.png`, and `merfish_10k_three_way_overlay.png`.
- **REQ-PLOT-003:** All plots must use `matplotlib.use('Agg')`, dpi=150, and the seaborn-v0_8-whitegrid style fallback.

### CLI — Command-Line Interface

- **REQ-CLI-001:** The script must accept `--phase baseline` and `--phase compare` as the sole required argument.
- **REQ-CLI-002:** All output must go to `tests/visual_eval/output/` by default.

### COMPAT — End-to-End Compatibility

- **REQ-COMPAT-001:** After running `--phase baseline`, the Rust test `export_merfish_init.rs` must be able to load `merfish_10k_graph.npz` and produce `merfish_10k_rust_init.npy` without errors.
- **REQ-COMPAT-002:** After running all three phases via `run_merfish_eval.sh`, the summary step must find and parse `merfish_10k_metrics.json` and print a PASS/FAIL line.
